databricks_cli/stack/api.py

```diff
@@ -0,0 +1,353 @@
+        from databricks when a GET request is called for it.
+        """
+        job_settings = resource_properties  # resource_properties of jobs are solely job settings.
+        click.echo("Deploying job '%s' with settings: \n%s \n" % (resource_id, json.dumps(
+            job_settings, indent=2, separators=(',', ': '))), nl=False)
+
+        if physical_id and 'job_id' in physical_id:
+            job_id = physical_id['job_id']
+            try:
+                self.update_job(job_settings, physical_id['job_id'])
+            except HTTPError:
+                # If updating a job fails, create the job with put_job
+                job_id = self.put_job(job_settings)
+        else:
+            job_id = self.put_job(job_settings)
+
+        job_url = "%s#job/%s" % (self.host, str(job_id))
+        click.echo("Job URL: %s" % job_url)
+        physical_id = {'job_id': job_id, "url": job_url}
+        deploy_output = self.jobs_client.get_job(job_id)
+        return physical_id, deploy_output
+
+    def deploy_resource(self, resource):  # overwrite to be added
+        """
+        Deploys a resource given a resource information extracted from the stack JSON configuration
+        template.
+
+        :param resource: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE and
+        RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        :return: dict resource_deploy_info- A dictionary of deployment information of the
+        resource to be stored at deploy time. It includes the resource id of the resource along
+        with the physical id and deploy output of the resource.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {'job_id': 123},
+        'timestamp': 123456789, 'deploy_output': {..}}
+        """
+        try:
+            resource_id = resource[RESOURCE_ID]
+            resource_service = resource[RESOURCE_SERVICE]
+            resource_properties = resource[RESOURCE_PROPERTIES]
+        except KeyError as e:
+            raise StackError("%s doesn't exist in resource config" % str(e))
+
+        # Deployment
+        physical_id = self._get_deployed_resource_physical_id(resource_id, resource_service)
+
+        if resource_service == JOBS_SERVICE:
+            physical_id, deploy_output = self.deploy_job(resource_id, resource_properties,
+                                                         physical_id)
+        else:
+            raise StackError("Resource service '%s' not supported" % resource_service)
+
+        resource_deploy_info = {RESOURCE_ID: resource_id, RESOURCE_SERVICE: resource_service,
+                                RESOURCE_DEPLOY_TIMESTAMP: datetime.now(),
+                                RESOURCE_PHYSICAL_ID: physical_id,
+                                RESOURCE_DEPLOY_OUTPUT: deploy_output}
+        return resource_deploy_info
+
+    def deploy(self, config_path):  # overwrite to be added
```

Comment:
Instead of passing in the `config_path`, let's pass in the JSON as either a `map` or `string` directly. Move the logic of how the file is read into cli.py. 
This will make your API client more reusable.

Reply:
Yeah, I like that idea. If you can create a separate method that takes a map/string, I would be ok with that.